L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h

```diff
@@ -32,15 +32,39 @@ class HGCalTriggerGeometryBase
         const std::string& name() const { return name_; } 
 
         const edm::ESHandle<CaloGeometry>& caloGeometry() const {return calo_geometry_;}
-        const HGCalGeometry* eeGeometry() const {return (static_cast<const HGCalGeometry*>(calo_geometry_->getSubdetectorGeometry(DetId::Forward,HGCEE)));}
-        const HGCalGeometry* fhGeometry() const {return (static_cast<const HGCalGeometry*>(calo_geometry_->getSubdetectorGeometry(DetId::Forward,HGCHEF)));}
-        const HcalGeometry* bhGeometry()  const {return (static_cast<const HcalGeometry*>(calo_geometry_->getSubdetectorGeometry(DetId::Hcal,HcalEndcap)));}
+        const HGCalGeometry* eeGeometry() const 
+        {
+            return ( hgc_ee_geometry_.isValid() ? hgc_ee_geometry_.product() : (static_cast<const HGCalGeometry*>(calo_geometry_->getSubdetectorGeometry(DetId::Forward,HGCEE))) );
+        }
+        const HGCalGeometry* fhGeometry() const 
+        {
+            return (hgc_hsi_geometry_.isValid() ? hgc_hsi_geometry_.product() : (static_cast<const HGCalGeometry*>(calo_geometry_->getSubdetectorGeometry(DetId::Forward,HGCHEF))) );
+        }
+        const HcalGeometry* bhGeometry()  const 
+        {
+            return (hgc_hsc_geometry_.isValid() ? nullptr : (static_cast<const HcalGeometry*>(calo_geometry_->getSubdetectorGeometry(DetId::Hcal,HcalEndcap))) );
```

Comment:
This is because in the V9 geometry the scintillator part of HGCAL is of type `HGCalGeometry` and not `HcalGeometry` as before.
But instead of returning a `nullptr` I could throw an exception if this function is called with the V9 geometry. And I would add a simple check function which tells if the geometry is V9 or V7/V8.

Reply:
Ah, right, the return type is different. Yes, in that case it might be better to have a more explicit protection against using this function with v9.